### PR TITLE
method for obtaining the roles/user from aws-auth configmap is added

### PR DESCRIPTION
**Pull request by ash-kamrip**
This is related to the issue no. #670 on the kubescape repo. I have added a method for obtaining the roles/user from aws-auth configmap

### CODE REVIEW

File: cloudsupport/v1/ekssupport.go [hunk 1]
```diff
@@ -19,17 +22,40 @@ type IEKSSupport interface {
...
 	GetRegion(cluster string) (string, error)
 	GetContextName(cluster string) string
+	GetAwsCfgMap(kapi *k8sinterface.KubernetesApi, namespace string) error
 }
 
 type EKSSupport struct {
 }
 
+// ListmapRoles will store mapped roles found inside the aws-auth
+var ListmapRoles []*MappedRoles
```
Comment (dwertent) [inline cloudsupport/v1/ekssupport.go:32]: Why are you using a global list? I think this list should be returned from the function

File: cloudsupport/v1/ekssupport.go [hunk 2]
```diff
@@ -107,3 +136,24 @@ func (eksSupport *EKSSupport) GetContextName(cluster string) string {
...
+// Getcfgmap returns the authdata containing mappings of iam-roles/groups or iam-users/groups
+func (EKSSupport *EKSSupport) GetAwsCfgMap(kapi *k8sinterface.KubernetesApi, namespace string) error {
+
+	resp, err := kapi.KubernetesClient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), awsauthconfigmap, metav1.GetOptions{})
+
+	if err != nil {
+		return fmt.Errorf("aws-auth config map not found")
+	}
+
+	if err := json.Unmarshal([]byte(resp.Data["mapRoles"]), &ListmapRoles); err != nil {
```
Comment (dwertent) [inline cloudsupport/v1/ekssupport.go:149]: You should define a struct with `mapRoles` and `mapUsers` and unmarshal the response object to that struct.

File: cloudsupport/v1/ekssupport.go [hunk 3]
```diff
@@ -107,3 +136,24 @@ func (eksSupport *EKSSupport) GetContextName(cluster string) string {
 	}
 	return ""
 }
+
+// Getcfgmap returns the authdata containing mappings of iam-roles/groups or iam-users/groups
+func (EKSSupport *EKSSupport) GetAwsCfgMap(kapi *k8sinterface.KubernetesApi, namespace string) error {
```
Comment (dwertent) [inline cloudsupport/v1/ekssupport.go:141]: The functions should return an EKS configMap

```suggestion
func (EKSSupport *EKSSupport) GetEKSCfgMap(kapi *k8sinterface.KubernetesApi, namespace string) (*EKSCfgMap ,error) {
```

**Review by dwertent — CHANGES_REQUESTED** [review]

File: cloudsupport/v1/ekssupport.go (hunk 3 above)
Comment (ash-kamrip) [reply cloudsupport/v1/ekssupport.go:141]: @dwertent ,can i use already available v1.configMap or should i create a new struct `EKSCfgMap`with same fields ?

File: cloudsupport/v1/ekssupport.go (hunk 2 above)
Comment (ash-kamrip) [reply cloudsupport/v1/ekssupport.go:149]: Ok will do this

**Comment by ash-kamrip** [comment]
@dwertent Hi David, I've made the requested changes , can you give it a look and tell me if further changes are required ?

File: cloudsupport/v1/ekssupport.go [hunk 4]
```diff
@@ -137,23 +136,25 @@ func (eksSupport *EKSSupport) GetContextName(cluster string) string {
...
+	eksCfgMap, err := kapi.KubernetesClient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), awsauthconfigmap, metav1.GetOptions{})
 
 	if err != nil {
-		return fmt.Errorf("aws-auth config map not found")
+		return nil, err
 	}
 
-	if err := json.Unmarshal([]byte(resp.Data["mapRoles"]), &ListmapRoles); err != nil {
-		return err
+	if err := json.Unmarshal([]byte(eksCfgMap.Data["mapRoles"]), &authData.MapRoles); err != nil {
```
Comment (dwertent) [inline cloudsupport/v1/ekssupport.go:150]: You should always validate the key is in the map before accessing it ->
```suggestion
if mapRoles, ok := eksCfgMap.Data["mapRoles"]; ok {
	if err := json.Unmarshal([]byte(mapRoles), &authData.MapRoles); err != nil {
			return nil, err
		}
	}		
}
```
You should apply a similar change on the `mapUsers` as well

**Review by dwertent — CHANGES_REQUESTED** [review]

File: cloudsupport/v1/ekssupport.go (hunk 4 above)
Comment (ash-kamrip) [reply cloudsupport/v1/ekssupport.go]: this is implemented

**Comment by ash-kamrip** [comment]
@dwertent pls check now, Implemented the checks for `mapUsers` and `mapRoles`

**Review by dwertent — APPROVED** [review]

**Pull request merged**